### ISSUE TITLE
Deploying error on cannot connect to domain.com

### ISSUE THREAD

**Issue by reksmeysrey** [lorisleiva/laravel-deployer#129]
When I run : php artisan deploy production 
it shows me an error cannot connect to domain.com

return [
    /*
    |--------------------------------------------------------------------------
    | Default deployment strategy
    |--------------------------------------------------------------------------
    |
    | This option defines which deployment strategy to use by default on all
    | of your hosts. Laravel Deployer provides some strategies out-of-box
    | for you to choose from explained in detail in the documentation.
    |
    | Supported: 'basic', 'firstdeploy', 'local', 'pull'.
    |
    */
    'default' => 'basic',
    /*
    |--------------------------------------------------------------------------
    | Custom deployment strategies
    |--------------------------------------------------------------------------
    |
    | Here, you can easily set up new custom strategies as a list of tasks.
    | Any key of this array are supported in the `default` option above.
    | Any key matching Laravel Deployer's strategies overrides them.
    |
    */
    'strategies' => [
        'staging' => [
            'hook:start',
            'deploy:prepare',
            'deploy:lock',
            'deploy:release',
            'deploy:update_code',
            'upload',
            'deploy:shared',
            'deploy:vendors',
            'copy_env:staging_value', // Perhatikan
            'deploy:writable',
            'hook:ready', // Di sini task artisan:generate berjalan
            'deploy:symlink',
            'deploy:unlock',
            'cleanup',
            'hook:done',
        ],
        'production' => [
            'hook:start',
            'deploy:prepare',
            'deploy:lock',
            'deploy:release',
            'deploy:update_code',
            'upload',
            'deploy:shared',
            'deploy:vendors',
            'copy_env:production_value', // Perhatikan
            'deploy:writable',
            'hook:ready', // Di sini task artisan:generate berjalan
            'deploy:symlink',
            'deploy:unlock',
            'cleanup',
            'hook:done',
        ]
    ],
    /*
    |--------------------------------------------------------------------------
    | Hooks
    |--------------------------------------------------------------------------
    |
    | Hooks let you customize your deployments conveniently by pushing tasks
    | into strategic places of your deployment flow. Each of the official
    | strategies invoke hooks in different ways to implement their logic.
    |
    */
    'hooks' => [
        // Right before we start deploying.
        'start' => [
            //
        ],
        // Code and composer vendors are ready but nothing is built.
        'build' => [
            'npm:install',
            'npm:production',
        ],
        // Deployment is done but not live yet (before symlink)
        'ready' => [
            'artisan:generate', // Task yang berasal dari my_recipe.php
            'artisan:storage:link',
            'artisan:view:clear',
            'artisan:cache:clear',
            'artisan:config:cache',
            'artisan:optimize',
            'artisan:migrate',
        ],
        // Deployment is done and live
        'done' => [
            'fpm:reload',
        ],
        // Deployment succeeded.
        'success' => [
            //
        ],
        // Deployment failed.
        'fail' => [
            //
        ],
    ],
    /*
    |--------------------------------------------------------------------------
    | Deployment options
    |--------------------------------------------------------------------------
    |
    | Options follow a simple key/value structure and are used within tasks
    | to make them more configurable and reusable. You can use options to
    | configure existing tasks or to use whithin your own custom tasks.
    |
    */
    'options' => [
        'application' => env('APP_NAME', 'Mabatama'),
        'repository' => 'git@gitlab.com:mabatama/mabatama.git',
        'php_fpm_service' => 'php7.2-fpm',
    ],
    /*
    |--------------------------------------------------------------------------
    | Hosts
    |--------------------------------------------------------------------------
    |
    | Here, you can define any domain or subdomain you want to deploy to.
    | You can provide them with roles and stages to filter them during
    | deployment. Read more about how to configure them in the docs.
    |
    */
    'hosts' => [
        'laravel-basic.com' => [
            'deploy_path' => '/var/www/laravel-basic',
            'user' => 'deployer',
        ],
        'test.laravel-basic.com' => [
            'deploy_path' => '/var/www/laravel-basic',
            'user' => 'deployer',
        ],
    ],
    /*
    |--------------------------------------------------------------------------
    | Localhost
    |--------------------------------------------------------------------------
    |
    | This localhost option give you the ability to deploy directly on your
    | local machine, without needing any SSH connection. You can use the
    | same configurations used by hosts to configure your localhost.
    |
    */
    'localhost' => [
        //
    ],
    /*
    |--------------------------------------------------------------------------
    | Include additional Deployer recipes
    |--------------------------------------------------------------------------
    |
    | Here, you can add any third party recipes to provide additional tasks,
    | options and strategies. Therefore, it also allows you to create and
    | include your own recipes to define more complex deployment flows.
    |
    */
    'include' => [
        'recipe/my_recipe.php', 
        'recipe/production_recipe.php',
        'recipe/staging_recipe.php',
    ],
    /*
    |--------------------------------------------------------------------------
    | Use a custom Deployer file
    |--------------------------------------------------------------------------
    |
    | If you know what you are doing and want to take complete control over
    | Deployer's file, you can provide its path here. Note that, without
    | this configuration file, the root's deployer file will be used.
    |
    */
    'custom_deployer_file' => false,
];

**Comment by lorisleiva** [comment]
It looks like you didn't configure the `stage` option on your hosts. E.g. `'stage' => 'production'`.

**Comment by reksmeysrey** [comment]
@lorisleiva Did you know example code or Ref for deploying to production? 
Sorry I'm just a student. Thanks

**Comment by reksmeysrey** [comment]
Run php aritsan deploy production 
ssh: connect to host domain.net port 21098: Network is unreachable

**Comment by Mxrck** [comment]
In your `deploy.php` your array "hosts" define the allowed hosts to use, in your case `laravel-basic.com` `test.laravel-basic.com`.

So, your command is:

`php artisan deploy laravel-basic.com`

or

`php artisan deploy test.laravel-basic.com`

**Comment by reksmeysrey** [comment]
Still have the same error

**Comment by lorisleiva** [comment]
Closing in favour of #130
